src/app/DTT/check_box/index.js

```diff
@@ -1,85 +1,122 @@
-import './styles.scss'
+import 'app/DTT/check_box/styles.scss'
+import { Checkbox as CCheckbox, Stack } from '@chakra-ui/react'
+import { append, compose, equals, includes, map, path, reject } from 'ramda'
+import { useEffect, useState } from 'react'
+import { useIsProductInternmatch, useIsProductLojing } from 'utils/helpers/check-product-name'
 
-import { Checkbox, FormControl, FormLabel, HStack } from '@chakra-ui/react'
-
-import { faCheckCircle } from '@fortawesome/free-solid-svg-icons'
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import { useGetAttributeFromProjectBaseEntity } from 'app/BE/project-be'
+import MandatorySymbol from 'app/layouts/components/form/mandatory-symbol'
+import { useSelector } from 'react-redux'
+import { selectCode } from 'redux/db/selectors'
 import { useIsFieldNotEmpty } from 'utils/contexts/IsFieldNotEmptyContext'
-import { useIsProductInternmatch } from 'utils/helpers/check-product-name'
 import useGetProductName from 'utils/helpers/get-product-name'
+import isJson from 'utils/helpers/is-json'
+import useProductColors from 'utils/productColors'
+import { isArray } from 'utils/helpers/is-type'
+import Select from 'app/DTT/select'
+import { useGetAttributeFromProjectBaseEntity } from 'app/BE/project-be'
 
-const Read = ({ data }) => {
-  return (
-    <HStack spacing={2}>
-      <Checkbox
-        colorScheme="green"
-        isChecked={data?.value === 'true'}
-        isDisabled="true"
-      >{`Yes`}</Checkbox>
-    </HStack>
-  )
-}
+const Read = ({ data, dataType }) => <Select.Read data={data} dataType={dataType} />
 
-const Write = ({ questionCode, data, onSendAnswer, isRequired, label }) => {
+const Write = ({
+  questionCode,
+  data,
+  onSendAnswer,
+  parentCode,
+  placeholderName,
+  mandatory,
+  config,
+}) => {
   const realm = useGetProductName().toLowerCase()
   const isProductInternmatch = useIsProductInternmatch()
+  const colorScheme = useGetAttributeFromProjectBaseEntity('PRI_SECONDARY_COLOR')?.valueString
+  let dataValueFromBackend = isJson(data?.value) ? JSON.parse(data.value) : data.value || ''
+  const isDataValueArray = Array.isArray(dataValueFromBackend)
+  const dataValue = isDataValueArray ? path([0])(dataValueFromBackend) : dataValueFromBackend
+  const isProductLojing = useIsProductLojing()
+  const [value, setValue] = useState(dataValue)
```

Comment:
format the order of initialization so that the state is on the top of the file after imports and every calculations and initializations can be done after that. 

ie: 
const [value, setValue] = useState(dataValue)
const isProductLojing = useIsProductLojing()
const isProductInternmatch = useIsProductInternmatch()
const realm = useGetProductName().toLowerCase()
const colorScheme = useGetAttributeFromProjectBaseEntity('PRI_SECONDARY_COLOR')?.valueString
.....

Reply:
Tried my best to do this however some things need to be declared first